### ISSUE TITLE
Matrix size mismatch

### ISSUE THREAD

**Issue by yuf0731** [Zerg-Overmind/GaussianFlow#12]
Hello, when I run: 
predicted_flow_by_gs = (cov_multi + proj_2D[gs_per_pixel_prev] - proj_2D_prev[gs_per_pixel_prev].detach() - x_mu_prev.permute(0, 2, 3, 1).detach()) * weight_per_gs_pixel_prev.detach()

there is a matrix mismatch problem, can you tell me how to fix it?

the error and debug information is:
![p2](https://github.com/user-attachments/assets/a4bb9f03-5805-499b-80c1-65e4b886578a)

**Comment by yuf0731** [comment]
I fixed the problem by the answer of #2 :)
